src/utils/url.js

```diff
@@ -216,6 +216,9 @@ export function parseUrl (url) {
 }
 
 export function isSupportedProtocol (url) {
+    if (typeof url !== 'string')
```

Comment:
I think that it is unnecessary.
We should always call `isSupportedProtocol` with `string` argument.